backend/app/providers/telegram.go

```diff
@@ -0,0 +1,65 @@
+package providers
+
+// This provider provides common logic, reused by Telegram auth and notifications verification
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	log "github.com/go-pkgz/lgr"
+
+	"github.com/umputun/remark42/backend/app/notify"
+)
+
+type tgRequester interface {
+	Request(ctx context.Context, method string, b []byte, data interface{}) error
+}
+
+// TGUpdatesReceiver used to dispatch telegram updates to multiple receivers
+type TGUpdatesReceiver interface {
+	ProcessUpdate(ctx context.Context, textUpdate string) error
```

Comment:
what do you think about adding Name or Id or something identifying the receiver? this way we could report more accurately "failure from blah" instead of not very helpful "failure from destination" error

Reply:
or maybe Name() method, unless you have planned to provide a functional adopter to this interface